### PR TITLE
Update ente-auth extension

### LINKED ISSUE #29272: [Ente Auth] Import skips TOTP entries when Ente export contains null OTP parameters
### Extension

https://www.raycast.com/chkpwd/ente-auth

### Raycast Version

0.67.1.0 (Beta)

### OS Version

macOS 26.3.1 (a) (25D771280a)

### Description

Hello,
I am using a fork of the Ente Auth Raycast extension and found an import issue that appears to affect Ente Auth exports containing literal null values in OTP URL parameters.

When I run the "Import Secrets" command, only 7 entries are imported even though my Ente Auth export contains 28 non-empty TOTP entries. The remaining 21 entries are silently skipped.

### Root cause found
The exported otpauth:// URLs for the skipped entries contain literal null values for standard OTP parameters, for example:

```text
period=null
```

or:

```text
algorithm=null&digits=null&period=null
```

The current parser in `src/helpers/secrets.ts` sanitizes some malformed Ente export data, but it does not normalize these null values before calling:

```ts
OTPAuth.URI.parse(url)
```

Because `period=null` and `digits=null` are not valid OTP parameter values, `OTPAuth.URI.parse` appears to throw. The import loop catches the error and skips that line:

```ts
try {
  const secret = parseSecretURL(line);
  if (secret) {
    secretsList.push(secret);
  }
} catch (error) {
  console.error("Error parsing line:", line, error);
}
```

This makes the import look successful, but only the successfully parsed subset is stored.

### Local diagnostic result

I ran a local diagnostic against my export file without printing secrets. The results were:

```json
{
  "totalLines": 29,
  "nonEmpty": 28,
  "looksImportable": 7,
  "trashed": 0,
  "failedBeforeOtpAuth": 0,
  "suspicious": 21
}
```

The 7 entries that looked importable had either valid OTP params or omitted optional params entirely.

The 21 skipped/suspicious entries had patterns like:

- `period=null`
- `algorithm=null`
- `digits=null`
- `algorithm=null&digits=null&period=null`

Examples of affected services in my export, without secrets:

- Tumblr: `period=null`
- Twitter: `period=null`
- Zoom: `period=null`
- Figma: `period=null`
- Facebook: `period=null`
- Instagram: `period=null`
- MEGA: `period=null`
- Discord: `algorithm=null&digits=null&period=null`
- Amazon: `algorithm=null&digits=null&period=null`
- Google: `algorithm=null&digits=null&period=null`
- Mozilla: `algorithm=null&digits=null&period=null`
- Twitch: `algorithm=null&digits=null&period=null`
- Reddit: `algorithm=null&digits=null&period=null`
- Zapier: `algorithm=null&digits=null&period=null`
- Microsoft: `algorithm=null&digits=null&period=null`
- Bitwarden: `algorithm=null&digits=null&period=null`
- Coursera: `algorithm=null&digits=null&period=null`

Why this should be safe to normalize

For TOTP, these parameters have common defaults:

- `algorithm`: `SHA1`
- `digits`: `6`
- `period`: `30`

If Ente exports `null` for these fields, that likely means the default value should be used, not that the entry is invalid.

### Suggested fix

In `sanitizeURL`, normalize null OTP params before parsing. For example, either remove them entirely so `otpauth` can apply defaults:

```ts
url = url.replace(/([?&])(algorithm|digits|period)=null(?=&|$)/g, "$1");
url = url.replace(/[?&]$/, "");
url = url.replace(/&&+/g, "&");
url = url.replace(/\?&/, "?");
```

Or replace them with explicit defaults:

```ts
url = url.replace(/([?&]algorithm=)null(?=&|$)/g, "$1SHA1");
url = url.replace(/([?&]digits=)null(?=&|$)/g, "$16");
url = url.replace(/([?&]period=)null(?=&|$)/g, "$130");
```

The first option may be cleaner because it lets the OTP parser apply defaults.

Also, `period` is currently stored from raw URL params:

```ts
period: getExtraInfo.get("period") ?? "",
```

It may be better to store the parsed/defaulted value instead, such as:

```ts
period: String(totp.period),
```

### Suggested tests

Add tests for Ente export URLs containing null params:

```ts
it("parses URLs with null period by using the default period", () => {
  const url = "otpauth://totp/user@example.com?secret=JBSWY3DPEHPK3PXP&issuer=GitHub&algorithm=SHA1&digits=6&period=null";
  const [secret] = parseSecrets([url]);

  expect(secret).toMatchObject({
    issuer: "GitHub",
    algorithm: "SHA1",
    digits: 6,
    period: "30",
  });
});

it("parses URLs with null algorithm, digits, and period by using defaults", () => {
  const url = "otpauth://totp/user@example.com?secret=JBSWY3DPEHPK3PXP&issuer=GitHub&algorithm=null&digits=null&period=null";
  const [secret] = parseSecrets([url]);

  expect(secret).toMatchObject({
    issuer: "GitHub",
    algorithm: "SHA1",
    digits: 6,
    period: "30",
  });
});
```

### Possible UX improvement

The import command currently only reports the number of successfully imported secrets. It would be helpful if it also reported skipped/failed entries, for example:

```text
7 secrets imported, 21 skipped due to parse errors
```

That would make this kind of issue much easier to detect.

Thanks for maintaining the extension.

### Steps To Reproduce

1. Have an Ente Auth account with multiple TOTP entries. 
2. Include entries whose exported otpauth:// URLs contain null OTP parameters, such as 'period=null' or 'algorithm=null&digits=null&period=null' 
3. Run the Raycast extension command: "Import Secrets". 
4. Check the import result toast. 
5. Observe that only the entries without null OTP parameters are imported. 
6. Inspect logs and notice parse errors for the skipped entries.

### Current Behaviour

- Export file contains 28 non-empty otpauth:// URLs. 
- Import command reports only 7 imported secrets. 
- No user-facing warning explains why the other entries were skipped. 
- The skipped entries are only visible as parse errors in console logs, if logs are inspected.

### Expected Behaviour

- All valid TOP entries from the Ente Auth export should import successfully. 
- Entries with missing optional/default OTP params should still import using normal defaults. 
- If an entry cannot be imported, the import command should ideally report how many entries failed or were skipped.

### PR DESCRIPTION
**Pull request by jakoss**
## Description

Fixes #29272 

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by chkpwd** [comment]
Hi @jakoss, I'll have a look.

Thank you.

**Review by pernielsentikaer — COMMENTED** [review]
Thanks for taking a look @chkpwd 🔥

File: extensions/ente-auth/src/helpers/secrets.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ const sanitizeURL = (url: string): string => {
 	// Fix bare issuer parameters (`&issuer&` → `&issuer=&`) emitted by some Ente Auth exports.
 	url = url.replace(/([?&]issuer)(?=&|$)/g, "$1=");
 
+	// Remove null OTP parameters so OTPAuth can apply its defaults.
+	url = url.replace(/([?&])(algorithm|digits|period)=null(?=&|$)/g, "$1");
```
[the remark this answers is not in the document]
Comment (chkpwd) [reply extensions/ente-auth/src/helpers/secrets.ts]: @jakoss could you verify this bit?

[the remark this answers is not in the document]
Comment (jakoss) [reply extensions/ente-auth/src/helpers/secrets.ts]: yes, will do

[the remark this answers is not in the document]
Comment (jakoss) [reply extensions/ente-auth/src/helpers/secrets.ts]: @chkpwd Fixed

**Review by pernielsentikaer — APPROVED** [review]
Looks good to me, approved 🔥

**Comment by thisguybikash** [comment]
Thank you @jakoss for fixing the Ente Auth import issue with `algorithm=null`, `digits=null`, and `period=null`.

I recently encountered this problem in practice: only 7 of my 28 TOTP entries were being imported into the Raycast extension. I investigated it, created a similar local fix, and then discovered while pulling the latest contributions that you had already identified and fixed the same issue.

Your change solved a real problem for me and saved me from having to recreate all of those TOTP entries manually. I especially appreciate that you included regression tests for the affected export formats. Thank you for contributing this fix!
